Src/SmtpServer/SmtpServerOptionsBuilder.cs

```diff
@@ -198,6 +212,13 @@ class SmtpServerOptions : ISmtpServerOptions
             /// The size of the buffer that is read from each call to the underlying network client.
             /// </summary>
             public int NetworkBufferSize { get; set; }
+
+            /// <summary>
+            /// Gets or sets the custom greeting message sent by the server in response to the initial SMTP connection.
+            /// This message is returned after the client connects and before any commands are issued (e.g., "220 mail.example.com v1.0 ESMTP ready").
+            /// If not set, a default greeting will be used.
+            /// </summary>
+            public string CustomGreetingMessage { get; set; }
```

Comment:
maybe this should actually be something like

Func<ISmtpContext, string> CustomGreetingMessage 

which would give the ability to make the message dynamic?